enclave/core/sgx/globals.c

```diff
@@ -7,6 +7,7 @@
 /* Note: The variables below are initialized during enclave loading */
 
 extern volatile const oe_sgx_enclave_properties_t oe_enclave_properties_sgx;
+volatile const oe_sgx_enclave_properties_t oe_enclave_properties_sgx_mutable;
```

Comment:
The memory properties (only the properties, not the actual amount of memory pages etc), are patched during enclave initialization to avoid patches to the base image, which would not be verifiable without having an actual copy of the base image. This means that we need a mutable version of `oe_enclave_properties_sgx`, which is initialized to a copy thereof, so that non-EEID enclaves won't see a difference.

Reply:
Maybe add that as a comment in the actual code?

Reply:
Done. I didn't consider this the final version yet, because I expected this to trigger more discussion. I've added it to the code too now.